src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+                        contigWithMultipleAlignments.toString());
+
+        final AlignmentInterval head = contigWithMultipleAlignments.alignmentIntervals.get(0),
+                                tail = contigWithMultipleAlignments.alignmentIntervals.get(contigWithMultipleAlignments.alignmentIntervals.size()-1);
+
+        // tail not resuming the "direction of flow" started by head
+        if ( !head.referenceSpan.getContig().equals(tail.referenceSpan.getContig()) || head.forwardStrand != tail.forwardStrand)
+            return true;
+
+        // reference span anomaly
+        final SimpleInterval referenceSpanHead = head.referenceSpan,
+                             referenceSpanTail = tail.referenceSpan;
+        if (referenceSpanHead.contains(referenceSpanTail) || referenceSpanTail.contains(referenceSpanHead))
+            return true;
+
+        final boolean refSpanAnomaly =
+                contigWithMultipleAlignments.alignmentIntervals
+                        .subList(1, contigWithMultipleAlignments.alignmentIntervals.size() - 1)
+                        .stream().map(ai -> ai.referenceSpan)
+                        .anyMatch( middle -> { // middle alignments' ref span either disjoint from head/tail, or completely contained in head/tail ref span
+                            final boolean badHead = middle.overlaps(referenceSpanHead) && !referenceSpanHead.contains(middle);
+                            final boolean badTail = middle.overlaps(referenceSpanTail) && !referenceSpanTail.contains(middle);
+                            return badHead || badTail;
+                        });
+        if (refSpanAnomaly)
+            return true;
+
+        if (head.forwardStrand) {
+            return referenceSpanHead.getStart() >= referenceSpanTail.getStart();
+        } else {
+            return referenceSpanHead.getEnd() <= referenceSpanHead.getEnd();
+        }
+    }
+
+    private static boolean isNonInformative(final Tuple3<Boolean, AlignedContig, List<String>> tigInfo) {
+        return tigInfo._1();
+    }
+
+    private static boolean hasOnly2Alignments(final Tuple3<Boolean, AlignedContig, List<String>> tigInfo) {
+        return tigInfo._2().alignmentIntervals.size() == 2;
+    }
+
+    /**
+     * Preprocess provided {@code originalConfiguration} of a particular contig and return a configuration after operation.
+     * Note that "original" is meant to be possibly different from the returned configuration,
+     * but DOES NOT mean the alignments of the contig as given by the aligner, i.e. the configuration should be
+     * one of the best given by {@link FilterLongReadAlignmentsSAMSpark#pickBestConfigurations(AlignedContig, Set, Double)}.
+     *
+     * Note that the configuration after this preprocessing may not even have the same number of alignment as in input configuration.
+     */
+    static Tuple2<List<AlignmentInterval>, List<AlignmentInterval>> fineTuneAlignments(final List<AlignmentInterval> originalConfiguration,
```

Comment:
I'd rename this method to something more specific -- `removeNonUniqueMappings`? -- and add some documentation on what the return value is.